src/uct/base/uct_mem.c

```diff
@@ -77,30 +78,29 @@ ucs_status_t uct_mem_alloc(void *addr, size_t min_length, unsigned flags,
     ssize_t huge_page_size;
 #endif
 
-    if (min_length == 0) {
-        ucs_error("Allocation length cannot be 0");
-        return UCS_ERR_INVALID_PARAM;
-    }
-
-    if (num_methods == 0) {
-        ucs_error("No allocation methods provided");
-        return UCS_ERR_INVALID_PARAM;
-    }
-
-    if ((flags & UCT_MD_MEM_FLAG_FIXED) &&
-        (!addr || ((uintptr_t)addr % ucs_get_page_size()))) {
-        ucs_debug("UCT_MD_MEM_FLAG_FIXED requires valid page size aligned address");
-        return UCS_ERR_INVALID_PARAM;
+    status = uct_mem_alloc_check_params(param);
+    if (status != UCS_OK) {
+        return status;
     }
 
-    for (method = methods; method < methods + num_methods; ++method) {
+    /* set defaults in case some param fields are not set */
+    address      = (param->field_mask & UCT_MEM_ALLOC_PARAM_FIELD_ADDR) ?
+                   param->address : NULL;
+    flags        = (param->field_mask & UCT_MEM_ALLOC_PARAM_FIELD_FLAGS) ?
+                   param->flags : 0;
+    mem_type     = (param->field_mask & UCT_MEM_ALLOC_PARAM_FIELD_MEM_TYPE) ?
+                   param->mem_type : UCS_MEMORY_TYPE_HOST;
+    alloc_length = param->length;
+
+    for (method = param->methods.methods;
+         method < param->methods.methods + param->methods.count; ++method) {
         ucs_trace("trying allocation method %s", uct_alloc_method_names[*method]);
 
         switch (*method) {
         case UCT_ALLOC_METHOD_MD:
             /* Allocate with one of the specified memory domains */
-            for (md_index = 0; md_index < num_mds; ++md_index) {
-                md = mds[md_index];
+            for (md_index = 0; md_index < param->mds.count; ++md_index) {
+                md = param->mds.mds[md_index];
```

Comment:
check that mds field is set

Reply:
Moving this check to uct_mem_alloc_check_params where I go through methods and if method_md found, error is returned if either FIELD_MDS isn't set or if mds.count < 1. Goal is that user doesn't use METHOD_MD without at least one valid entry and with FIELD_MDS set.

Reply:
simpler to check here and skip METHOD_MD is not set

Reply:
I agree it's simpler but we've been consolidating basic checks and invalid parameter usage in uct_mem_alloc_check_params and handling only special cases in uct_mem_alloc. Is it ok to leave the check https://github.com/openucx/ucx/commit/add6339434cc29e5ee635c89e5bba5d79fc9d8f8#diff-6282d9e471573ebace2884f6debada46R422 as is?